controller/index.js

```diff
@@ -1,1019 +1,1054 @@
+   * @private
+   */
+  controller._requestNewFile = function (params) {
+    var res = params.res
+    var itemJson = params.itemJson
+
+    if (params.err) {
+      return res.status(params.err.code || 400).send(params.err.error || params.err)
+    }
+
+    // flatten the data from an array to sep objects/arrays
+    var itemData
+    var itemFeatures
+    if (itemJson && itemJson.data && itemJson.data[0]) {
+      itemData = itemJson.data[0]
+    }
+
+    if (itemData && itemData.features) {
+      itemFeatures = itemData.features
+    }
+
+    if (!itemFeatures || !itemFeatures.length) {
+      return res.status(404).send('No features exist for the requested FeatureService layer')
+    }
+
+    // this logic sure does suck...
+    var name = (itemJson && itemJson.data && itemJson.data[0] && (itemJson.data[0].name || (itemJson.data[0].info && itemJson.data[0].info.name))) ? itemJson.data[0].name || itemJson.data[0].info.name : itemJson.name || itemJson.title
+    // cleanze the name
+    name = name.replace(/\/|,|&\|/g, '').replace(/ /g, '_').replace(/\(|\)|\$/g, '')
+    name = (name.length > 150) ? name.substr(0, 150) : name
+    params.name = name
+
+    if ((itemJson.koop_status && itemJson.koop_status === 'too big') || agol.forceExportWorker) {
+      return controller._exportLarge(params)
+    }
+
+    if (itemJson && itemJson.data && itemJson.data[0]) {
+      return controller._exportToFormat(params)
+    }
+
+    return res.status(400).send('Could not create export, missing data')
+  }
+
+  /**
+   * Exports a large dataset
+   * calls the exportLarge method on the agol model
+   * @param {objects} params - file export parameters
+   */
+  controller._exportLarge = function (params) {
+    var req = params.req
+    var res = params.res
+    var itemJson = params.itemJson
+    req.query.name = params.name
+
+    // set the geometry type so the exporter can do its thing for csv points (add x,y)
+    if (itemJson.data && itemJson.data && itemJson.data.info && itemJson.data.info.geometryType) {
+      req.query.geomType = itemJson.data[0].info.geometryType
+    }
+
+    // force export of large data
+    req.query.large = true
+
+    agol.exportFile(params, req.query, function (err, result) {
+      if (err) {
+        return res.status(err.code || 400).send(err)
       }
+
+      if (result && result.status && result.status === 'processing') {
+        return controller._returnProcessing(req, res, itemJson)
+      }
+
+      if (req.query.url_only) {
+        // reuse this code...
+        var origUrl = req.originalUrl.split('?')
+        origUrl[0] = origUrl[0].replace(/json/, req.params.format)
+
+        var newUrl = req.protocol + '://' + req.get('host') + origUrl[0] + '?' + origUrl[1]
+          .replace(/url_only=true&|url_only=true|/, '')
+          .replace('format=' + req.params.format, '')
+          .replace('&format=' + req.params.format, '')
+
+        return res.json({url: newUrl})
+      }
+      res.contentType('text')
```

Comment:
This actually never gets called. when calling exportFile with a "query.large = true" the response will always return as processing 202. Going to remove this.